src/Swarrot/Processor/XDeath/XDeathMaxLifetimeProcessor.php

```diff
@@ -0,0 +1,186 @@
+use Swarrot\Broker\Message;
+use Swarrot\Broker\MessageProvider\MessageProviderInterface;
+use Swarrot\Processor\ProcessorInterface;
+use Swarrot\Processor\ConfigurableInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class XDeathMaxLifetimeProcessor implements ConfigurableInterface
+{
+    /**
+     * @var ProcessorInterface
+     */
+    protected $processor;
+
+    /**
+     * @var MessageProviderInterface
+     */
+    private $messageProvider;
+
+    /**
+     * @var \Closure
+     */
+    private $callback;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param ProcessorInterface $processor
+     * @param MessageProviderInterface $messageProvider
+     * @param \Closure $callback
+     * @param LoggerInterface|null $logger
+     */
+    public function __construct(
+        ProcessorInterface $processor,
+        MessageProviderInterface $messageProvider,
+        \Closure $callback,
+        LoggerInterface $logger = null
+    )
+    {
+        $this->processor = $processor;
+        $this->messageProvider = $messageProvider;
+        $this->callback = $callback;
+        $this->logger = $logger ?: new NullLogger();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setDefaultOptions(OptionsResolver $resolver)
+    {
+        $resolver
+            ->setDefaults(array(
+                'x_death_max_lifetime' => 3600,
+                'x_death_max_lifetime_log_levels_map' => array(),
+                'x_death_max_lifetime_fail_log_levels_map' => array(),
+            ))
+            ->setAllowedTypes('x_death_max_lifetime', 'int')
+            ->setAllowedTypes('x_death_max_lifetime_log_levels_map', 'array')
+            ->setAllowedTypes('x_death_max_lifetime_fail_log_levels_map', 'array');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(Message $message, array $options)
+    {
+        try {
+            return $this->processor->process($message, $options);
+        } catch (\Throwable $e) {
+            return $this->handleException($e, $message, $options);
+        } catch (\Exception $e) {
+            return $this->handleException($e, $message, $options);
+        }
+    }
+
+    /**
+     * @param $exception
+     * @param Message $message
+     * @param array $options
+     *
+     * @return mixed
+     */
+    private function handleException($exception, Message $message, array $options)
+    {
+        $headers = $message->getProperties();
+        if (isset($headers['headers']['x-death'], $options['x_death_max_lifetime'])) {
+            $xDeathHeaders = $headers['headers']['x-death'];
+            // PhpAmqpLib compatibility
+            if (class_exists('PhpAmqpLib\Wire\AMQPArray') && $headers['headers']['x-death'] instanceof AMQPArray) {
+                $xDeathHeaders = $headers['headers']['x-death']->getNativeData();
+            }
+            $queueName = $this->messageProvider->getQueueName();
```

Comment:
why injecting the whole MessageProvider just to get the queue name ? You could inject just the queue name